lib/si-layer-cache/src/db.rs

```diff
@@ -118,65 +102,89 @@ where
             encrypted_secret::CACHE_NAME,
             pg_pool.clone(),
-            memory_cache_config.clone(),
-            disk_cache_config.clone(),
+            cache_config
+                .clone()
+                .with_memory_percentage(0.05)
+                .with_disk_capacity(8 * GIGABYTES)
+                .with_path_join(encrypted_secret::CACHE_NAME),
             compute_executor.clone(),
-        )?;
+            tracker.clone(),
+            token.clone(),
+        )
+        .await?;
 
-        let func_run_cache: LayerCache<Arc<FuncRun>> = LayerCache::new(
+        let func_run_cache: Arc<LayerCache<Arc<FuncRun>>> = LayerCache::new(
             func_run::CACHE_NAME,
             pg_pool.clone(),
-            memory_cache_config.clone(),
-            disk_cache_config.clone(),
+            cache_config
+                .clone()
+                .with_memory_percentage(0.05)
+                .with_disk_capacity(8 * GIGABYTES)
+                .with_path_join(func_run::CACHE_NAME),
             compute_executor.clone(),
-        )?;
+            tracker.clone(),
+            token.clone(),
+        )
+        .await?;
 
-        let func_run_log_cache: LayerCache<Arc<FuncRunLog>> = LayerCache::new(
+        let func_run_log_cache: Arc<LayerCache<Arc<FuncRunLog>>> = LayerCache::new(
             func_run_log::CACHE_NAME,
             pg_pool.clone(),
-            memory_cache_config.clone(),
-            disk_cache_config.clone(),
+            cache_config
+                .clone()
+                .with_memory_percentage(0.05)
+                .with_disk_capacity(8 * GIGABYTES)
+                .with_path_join(func_run_log::CACHE_NAME),
             compute_executor.clone(),
-        )?;
+            tracker.clone(),
+            token.clone(),
+        )
+        .await?;
 
-        let rebase_batch_cache: LayerCache<Arc<RebaseBatchValue>> = LayerCache::new(
+        let rebase_batch_cache: Arc<LayerCache<Arc<RebaseBatchValue>>> = LayerCache::new(
             rebase_batch::CACHE_NAME,
             pg_pool.clone(),
-            memory_cache_config.clone(),
-            disk_cache_config.clone(),
+            cache_config
+                .clone()
+                .with_memory_percentage(0.05)
+                .with_disk_capacity(8 * GIGABYTES)
+                .with_path_join(rebase_batch::CACHE_NAME),
             compute_executor.clone(),
-        )?;
+            tracker.clone(),
+            token.clone(),
+        )
+        .await?;
 
-        let snapshot_cache: LayerCache<Arc<WorkspaceSnapshotValue>> = LayerCache::new(
+        let snapshot_cache: Arc<LayerCache<Arc<WorkspaceSnapshotValue>>> = LayerCache::new(
             workspace_snapshot::CACHE_NAME,
             pg_pool.clone(),
-            memory_cache_config.clone(),
-            disk_cache_config.clone(),
+            cache_config
+                .clone()
+                .with_memory_percentage(0.50)
```

Comment:
A good solution before working out how to use a single cache for everything

Reply:
Agreed!